### PR TITLE
Quiche roll 20230713142838

### PR DESCRIPTION
**Pull request by vanessareimer**
https://github.com/google/quiche/compare/5cc27c4cf..68e7ffc3a

```
$ git log 5cc27c4cf..68e7ffc3a --date=short --no-merges --format="%ad %al %s"

2023-07-13 bnc Remove IsSupportedTransportVersion().
2023-07-13 birenroy Removes a DCHECK that can be reached by fuzzing.
2023-07-13 wub Internal change
2023-07-12 wub Let QuicReceivedPacketManager trim ack ranges when packet is received, instead of when sending acks.
2023-07-12 haoyuewang Log all data in the sequencer when QUIC_HTTP_INVALID_FRAME_SEQUENCE_ON_SPDY_STREAM happens with no bytes ever consumed.
2023-07-11 quiche-dev Fixing tests in anonymous_tokens_rsa_bssa_client_test.cc.
2023-07-11 quiche-dev When the GFE issues a 413 error due to excessively long request headers, also send a Clear-Site-Data: "cookies" response header, but only if the portion of the cookie header we have is greater than a certain length.
2023-07-11 diannahu Internal change
2023-07-09 vasilvv Add a connection option (`kBSUS`) to force the server to buffer incoming streams until the SETTINGS frame is received.
2023-07-07 wub Internal change
2023-07-06 wub In QuicCryptoClientConfig: - Change session_cache_ from std::unique_ptr to std::shared_ptr. This allows multiple QuicCryptoClientConfig(s) to use the same Session Cache. - Add set_session_cache(). So a caller can change Session Cache after construction. - Remove mutable_session_cache(). It is the same as session_cache().
2023-07-06 quiche-dev Updating RsaBlindSignatureVerify in crypto_utils to use RSA struct only when verifying even for public metadata signature verification.
```

Risk Level: Low
Testing: Existing tests pass
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: test/common/quic/envoy_quic_client_stream_test.cc [hunk 1]
```diff
@@ -317,7 +317,7 @@ TEST_F(EnvoyQuicClientStreamTest, PostRequestAnd1xx) {
   size_t i = 0;
   // Receive several 10x headers, only the first 100 Continue header should be
   // delivered.
-  for (const std::string& status : {"100", "199", "100"}) {
+  for (const std::string status : {"100", "199", "100"}) {
```
Comment (htuch) [inline test/common/quic/envoy_quic_client_stream_test.cc:320]: Why is this necessary?

Comment (vanessareimer) [reply test/common/quic/envoy_quic_client_stream_test.cc]: Reverted. Was pointed out to me that this was a gcc vs clang build thing.

**Review by htuch — APPROVED** [review]
